node/vault.ts

```diff
@@ -6,6 +6,8 @@ import crypto = require('crypto');
 
 var uuidV4 = require('uuid/v4');
 var algorithm = "aes-256-ctr";
+var encryptEncoding: 'hex' = 'hex';
+var unencryptedEncoding: 'utf8' = 'utf8';
```

Comment:
lgtm

- let's move the duplicate logic into `getKey`
- check w/ CJ and let's use sha512
- make sure it works on node 5, otherwise fallback to the old logic?

Reply:
- and discuss with cj the difference between the hash approach and scrypt. I think the hash approach is sufficient based on our usage. I might be worth doing a small perf test between the two. For example, run in a loop N times and measure the total time; repeat several times to make sure the results are consistent.